protocol/contracts/core/OtcOffer.sol

```diff
@@ -0,0 +1,168 @@
+        address _userSellToken,
+        address _userBuyToken,
+        address _fundsOwner,
+        uint256 _offerPrice,
+        OfferPricingToken _offerPricingToken,
+        uint256 _offerPriceUpdateThresholdBps
+    ) {
+        userSellToken = IERC20Metadata(_userSellToken);
+        userBuyToken = IERC20Metadata(_userBuyToken);
+        fundsOwner = _fundsOwner;
+
+        offerPrice = _offerPrice;
+        offerPricingToken = _offerPricingToken;
+        offerPriceUpdateThresholdBps = _offerPriceUpdateThresholdBps;
+
+        // The price is always specified in 18dp
+        // Eg If selling OHM (9dp) for USDC (6dp):
+        // 1000 OHM (9dp) * 11 USDC/OHM (18dp) = 11_000 USDC (6dp)
+        // So this would need to get scaled down by 30dp
+        uint256 scaleDecimals = offerPricingToken == OfferPricingToken.UserBuyToken
+            ? OFFER_PRICE_DECIMALS + userSellToken.decimals() - userBuyToken.decimals()
+            : OFFER_PRICE_DECIMALS + userBuyToken.decimals() - userSellToken.decimals();
+        scalar = 10 ** scaleDecimals;
+    }
+
+    /// @notice Owner can pause user swaps from occuring
+    function pause() external onlyOwner {
+        _pause();
+    }
+
+    /// @notice Owner can unpause so user swaps can occur
+    function unpause() external onlyOwner {
+        _unpause();
+    }
+
+    /// @notice Owner can update where the funds are pulled from and sent to upon a swap
+    function setFundsOwner(address _fundsOwner) external onlyOwner {
+        if (_fundsOwner == address(0)) revert CommonEventsAndErrors.InvalidAddress();
+        fundsOwner = _fundsOwner;
+        emit FundsOwnerSet(_fundsOwner);
+    }
+
+    /// @notice Owner can update the offer price at which user swaps can occur
+    /// @dev The new price must be within the `offerPriceUpdateThresholdBps` threshold of
+    /// the existing price
+    function setOfferPrice(uint256 _offerPrice) external onlyOwner {
+        if (_offerPrice == 0) revert CommonEventsAndErrors.ExpectedNonZero();
+
+        uint256 existingOfferPrice = offerPrice;
+        uint256 absMaxDelta = existingOfferPrice * offerPriceUpdateThresholdBps / BPS_100_PCT;
+        if (_offerPrice > existingOfferPrice) {
+            uint256 maxValid = existingOfferPrice + absMaxDelta;
+            if (_offerPrice > maxValid) revert OfferPriceDeltaTooBig(_offerPrice, maxValid);
+        } else {
+            uint256 minValid = existingOfferPrice > absMaxDelta
+                ? existingOfferPrice - absMaxDelta
+                : 0;
+            if (_offerPrice < minValid) revert OfferPriceDeltaTooBig(_offerPrice, minValid);
+        }
+
+        offerPrice = _offerPrice;
+        emit OfferPriceSet(_offerPrice);
+    }
+
+    /// @notice Owner can update the threshold for when updating the offer price
+    function setOfferPriceUpdateThresholdBps(uint256 _offerPriceUpdateThresholdBps) external onlyOwner {
+        offerPriceUpdateThresholdBps = _offerPriceUpdateThresholdBps;
```

Comment:
Should this have a check < BPS_100_PCT ?

Reply:
I'm not sure about that - what if current price is $2 and you want to set to $6

That's 300%

Reply:
Can do that yep